espn_api/baseball/box_score.py

```diff
@@ -0,0 +1,52 @@
+from .constant import STATS_MAP
+
+class BoxScore(object):
+    ''' '''
+    def __init__(self, data):
+        self.winner = data['winner']
+        
+        home_team = self._process_team(data['home'])
+
+        self.home_team = home_team['id']

```

Comment:
For box score, you should add home and away team scores for the matchup.

Reply:
I have `home_wins`, `home_losses`, and `home_ties`, and `away_wins`, `away_losses`, and `away_ties` defined below - or did you mean something else by the team scores?

Reply:
The wins and losses are for the teams overall record but box_score is getting information about a specific matchup against two teams. The home and away team scores shows the score between the two teams for the matchup.

For example this is the matchup/box_score for a league I am in:  https://fantasy.espn.com/baseball/boxscore?leagueId=81134470&matchupPeriodId=6&scoringPeriodId=46&seasonId=2021&teamId=6&view=matchup. In this case the scores would be 433 and 295.

Reply:
Ah I see. So the league I've been looking at (my own) is a categories head to head league - so the wins and losses are for the matchup only, and not for the season. For example, in my case there are 12 categories, so the home score could be 6-5-1 and the away score would be 5-6-1.

I don't play in any points head to head leagues so I don't know how the data is represented in that case. How do you suggest we represent the different kinds of leagues?

Reply:
I'm thinking we use what's in the league settings (for me it's `['settings']['scoringSettings']['scoringType'] == 'H2H_CATEGORY'` and have different `BoxScore` subclasses based on the league type.

Reply:
Oh that makes a lot of sense with the wins/losses/ties information in the box score. 

Yeah thats a great idea to use the league scoring setting. I have this same issue with basketball league as well for categories head to head scoring. I think making some generic sub box_score classes that all sports can use would be great and then can be set when the league is initialized.

I will create a ticket for this work, but in that case I think this PR is good!

Reply:
I took a stab at making the baseball `BoxScore` a base class and having the baseball league set the boxscore subclass on instantiation - probably is portable for other sports (but I'm not currently in any ESPN leagues for other sports so I can't test a more general solution).

Reply:
Thanks this is a great start and I can definitely work on expanding this to other sports! I appreciate all of your work!